islandora_plupload.module

```diff
@@ -183,3 +189,30 @@ function islandora_plupload_file_save($file) {
   unset($file->original);
   return $file;
 }
+
+/**
+ * Get a list of available extensions for the object in context.
+ *
+ * @return string
+ *   List of allowable extensions for the object's datastrams.
+ */
+function islandora_plupload_get_available_extensions() {
+  module_load_include('inc', 'islandora', 'includes/mimetype.utils');
+  $extensions = "";
+  $object = menu_get_object('islandora_object', 2);
+  if (isset($object) && isset($object->models)) {
+    $models = $object->models;
+    // Remove the fedora-system:FedoraObject-3.0 model.
+    array_pop($models);
+    $dsids = islandora_get_datastreams_requirements_from_models($models);
+
+    foreach ($dsids as $dsid) {
+      $extensions .= implode(" ", islandora_get_extensions_for_datastream($object, $dsid['id']));
+      $extensions .= " ";
+
+    }
+    // XXX Module disables acceptable file types without xls ext added.
```

Comment:
This doesn't seem to be true; I don't see any difference in functionality with the next line commented out. It also breaks the convention we're establishing here that a datastream must have a mimetype associated from its DS composite model. Either way, there should be a better way of doing this.